lib/spack/spack/database.py

```diff
@@ -393,7 +393,7 @@ def _add(self, spec, path, directory_layout=None):
 
         else:
             self._data[key] = InstallRecord(spec, path, True)
-            for dep in spec.dependencies.values():
+            for dep in spec.dependencies(spack.DEP_INSTALL):
```

Comment:
Yeah, adding `DEP_RUN` should be done here. Still not convinced that build dependencies need to be kept around, but I'll do so and add a comment.